wamr-compiler/CMakeLists.txt

```diff
@@ -387,6 +387,12 @@ if (NOT MSVC)
       target_link_libraries (wamrc ssp.a ws2_32)
   else()
       target_link_libraries (wamrc -ldl)
+      # Link libc++ statically to reduce the runtime dependency
+      target_link_libraries (wamrc -static-libstdc++)
+      # If not on macOS, link libgcc statically
+      if (NOT APPLE)
+        target_link_libraries (wamrc -static-libgcc)
```

Comment:
Can we make this configurable?
I think static link such libraries is not a very common requirement.